### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by odow**
Updates for https://github.com/jump-dev/JuMP.jl/pull/3309

### CODE REVIEW

**Comment by mzy2240** [comment]
I am using AmplNLWriter with gurobi, cplex, xpress, and copt, and all seem to work fine. The current README feels a little bit misleading in terms of solver support.

**Comment by odow** [comment]
I pushed a small change. I'm a bit hesitant to explicitly call out the commercial solvers. Any suggestions for what we could change without mentioning a particular solver?

**Comment by odow** [comment]
> gurobi, cplex, xpress, and copt

The other reason is that these all have their own Julia packages, so there is limited benefit to using AmplNLWriter.

**Comment by mzy2240** [comment]
> The other reason is that these all have their own Julia packages, so there is limited benefit to using AmplNLWriter.

I did not find a way to use dedicated Julia package to work with the AMPL-bundled commercial solvers. IMHO that's the only constraint to stick to AmplNLWriter. In terms of the README, I would say it might be easier to simply add "Other AMPL-compatible solvers are also supported, though we recommend to use their own Julia packages" in the Solvers section.

**Comment by odow** [comment]
> I did not find a way to use dedicated Julia package to work with the AMPL-bundled commercial solvers. 

I think this means that your solver licenses are tied to the AMPL binaries, and that you can't use other interfaces.

Using the binaries JuMP and AmplNLWriter.jl might be a violation of the license. You should check your license conditions, and/or contact AMPL for advice.  

Note that AmplNLWriter is not an official product of AMPL Optimization. Instead, we use their open-source ASL driver to access compatible solvers: https://github.com/ampl/asl.

File: README.md [hunk 1]
```diff
@@ -4,85 +4,106 @@
...
-**Note: AmplNLWriter requires Julia 1.6 or later.**
-
-### Solvers
+## Use with JuMP
 
-You also need an AMPL compatible solver.
+AmplNLWriter requires an AMPL compatible solver binary to function.
 
-#### Bonmin (https://github.com/coin-or/Bonmin)
+Pass a string pointing to an AMPL-compatible solver binary as the first
```
Comment (odow) [inline README.md:29]:
```suggestion
Pass a string pointing to any AMPL-compatible solver binary as the first
```

**Comment by mzy2240** [comment]
> Using the binaries JuMP and AmplNLWriter.jl might be a violation of the license. You should check your license conditions, and/or contact AMPL for advice.

Thank you for the notice. I consulted with them about using JuMP with AMPL, and I am sure my license covers the this type of usage, academically and commercially.

**Comment by odow** [comment]
> I consulted with them about using JuMP with AMPL, and I am sure my license covers this type of usage, academically and commercially.

Cool. Just checking. 

Is the proposed switch from "an" to "any" enough?

**Comment by mzy2240** [comment]
> Is the proposed switch from "an" to "any" enough?

LGTM lol.

**Pull request merged**